django/db/models/sql/query.py

```diff
@@ -260,10 +261,27 @@ def __str__(self):
         substituted in (use sql_with_params() to see the unsubstituted string).
 
         Parameter values won't necessarily be quoted correctly, since that is
-        done by the database interface at execution time.
+        done by the database interface at execution time; rudimentary quoting
+        is performed below.
         """
         sql, params = self.sql_with_params()
-        return sql % params
+
+        unquoted_param_types = (int, float, Decimal)
+        quoted_params = []
+
+        for param in params:
+            if type(param) in unquoted_param_types:
+                quoted_params.append(param)
+            else:
+                if type(param) is str:
+                    param = param.replace("'", "''")
+
+                quoted_params.append("'{param}'".format(
+                    param=param,
+                ))
+        sql = sql.replace(" ESCAPE '\\'", "")
```

Comment:
What does this `ESCAPE` do? I believe removing it is causing several tests in the suite to fail. I have this here, for now, to produce SQL that can be run.

Reply:
It looks like it's from:
https://github.com/django/django/blob/c1c68d1ac0f0d50eb37df32892b132f31a1179da/django/db/backends/sqlite3/base.py#L101-L103
I guess that reveals one problem with expecting you can get SQL to run using this method (if you have a mixed multi-database configuration). `sql_with_params()` is called with `DEFAULT_DB_ALIAS`.

I don't have much expertise here but are we basically "reimplementing the full quoting behavior of a DB cursor" (or a simpler version) as mentioned in https://code.djangoproject.com/ticket/12611#comment:1.